### PR TITLE
CLOUDP-48516: Add TLS support

### PR DESCRIPTION
**Pull request by fabianlindfors**
With this PR we add support for TLS to the broker API server. Users can optionally pass a certificate and private key file with environment variables.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -92,17 +92,27 @@ func startBrokerServer() {
...
 	// Start broker HTTP server.
-	endpoint := host + ":" + strconv.Itoa(port)
-	if err = http.ListenAndServe(endpoint, nil); err != nil {
+	address := host + ":" + strconv.Itoa(port)
+
+	var serverErr error
+	if tlsEnabled {
+		serverErr = http.ListenAndServeTLS(address, tlsCertFile, tlsKeyFile, router)
+	} else {
+		serverErr = http.ListenAndServe(address, router)
```
Comment (BenElgar) [inline main.go]: Can we log a warning if TLS is disabled?

File: main.go [hunk 2]
```diff
@@ -92,17 +92,27 @@ func startBrokerServer() {
 	baseURL := strings.TrimRight(getEnvOrDefault("ATLAS_BASE_URL", DefaultAtlasBaseURL), "/")
 	router.Use(atlasbroker.AuthMiddleware(baseURL))
 
-	// Mount broker server at the root.
-	http.Handle("/", router)
+	// Configure TLS from environment variables.
+	tlsCertFile := getEnvOrDefault("BROKER_TLS_CERT_FILE", "")
+	tlsKeyFile := getEnvOrDefault("BROKER_TLS_KEY_FILE", "")
+	tlsEnabled := tlsCertFile != "" && tlsKeyFile != ""
```
Comment (BenElgar) [inline main.go]: Can we exit if only one is defined?

File: main.go (hunk 1 above)
Comment (fabianlindfors) [reply main.go]: Sure! Done

File: main.go (hunk 2 above)
Comment (fabianlindfors) [reply main.go]: Yes!

**Comment by fabianlindfors** [comment]
I've added documentation to the dev README for setting up TLS. I was considering adding samples for TLS setup but unfortunately the service catalog only accept a CA bundle literal. As it can't be included from a secret the user needs to update `service-broker.yaml` manually.

File: main.go [hunk 3]
```diff
@@ -92,21 +92,46 @@ func startBrokerServer() {
 	baseURL := strings.TrimRight(getEnvOrDefault("ATLAS_BASE_URL", DefaultAtlasBaseURL), "/")
 	router.Use(atlasbroker.AuthMiddleware(baseURL))
 
-	// Mount broker server at the root.
-	http.Handle("/", router)
+	// Configure TLS from environment variables.
+	tlsEnabled, tlsCertFile, tlsKeyFile := getTLSConfig(logger)
```
Comment (BenElgar) [inline main.go]: Sounds like it's getting back a file object or contents rather than path.

**Pull request merged**